ddclient.in

```diff
@@ -2146,6 +2154,128 @@ sub geturl {
     return $reply;
 }
 
+######################################################################
+## geturl -- via curl
+######################################################################
+sub curl_cmd {
+    my $params = join("\n",@_);
+    my $tmpfile;
+
+    return '' if (length($params) == 0); ## no parameters provided
+
+    # debug("CURL:\r\n%s\r\n",$params); ## Uncomment only if required as may expose credentials
+
+    eval "no warnings 'all'; use File::Spec;";
+    $tmpfile = ((!$@) ? File::Spec->tmpdir() : '/tmp').'/ddclient-';
+    my @chars = ("A".."Z", "a".."z", "0".."9");
+    $tmpfile .= $chars[rand @chars] for 1..10;
+    debug("CURL Tempfile: %s",$tmpfile);
+    open(my $TFH, '>', $tmpfile) or fatal("Falied to open $tmpfile for curl");
+    chmod(0600,$tmpfile);
+    print($TFH $params);
+    close($TFH);
+    my $reply = qx{ curl --config $tmpfile 2>/dev/null };
+    unlink($tmpfile);
+    return $reply;
+}
+
+sub fetch_via_curl {
+    my ($params) = @_;
+    my $proxy = $params->{proxy} // '';
+    my $url = $params->{url} // '';
+    my $login = $params->{login} // '';
+    my $password = $params->{password} // '';
+    my $ipversion = $params->{ipversion} // '';
+    my $headers = $params->{headers} // '';
+    my $method = $params->{method} // 'GET';
+    my $data = $params->{data} // '';
+
+    my $reply;
+    my $server;
+    my $use_ssl = 0;
+    my $force_ssl = 0;
+    my $protocol;
+    my $timeout  = opt('timeout');
+    my @curlopt = ();
+    my @header_lines = ();
+    my $is_get_ip = (((caller(2))[3] // '') eq 'ddclient::get_ip');
+
+    if (!`curl --version 2>/dev/null`) {
+        fatal("You requested network access with curl but curl is not installed.");
+    }
+
+    ## canonify proxy and url
+    $force_ssl = 1 if ($url =~ /^https:/);
+    $proxy  =~ s%^https?://%%i;
+    $url    =~ s%^https?://%%i;
+    $server = $url;
+    $server =~ s%/.*%%;
+    $url    = "/" unless $url =~ m%/%;
+    $url    =~ s%^[^/]*/%%;
+
+    debug("opt(fw = %s)", opt('fw')) if opt('fw');
+    debug("glo fw = %s", $globals{'fw'}) if $globals{'fw'};
+    ## always omit SSL for connections to local router
+    $use_ssl = 1 if ($force_ssl || ($globals{'ssl'} && !$is_get_ip));
+
+    $protocol = ($use_ssl ? "https" : "http");
+
+    debug("proxy    = %s", $proxy);
+    debug("protocol = %s", $protocol);
+    debug("server   = %s", $server);
+    debug("url      = %s", $url);
+    debug("ip ver   = %s", $ipversion);
+
+    if (!opt('exec')) {
+        debug("skipped network connection");
+        verbose("SENDING:", "%s", "${server}/${url}");
+    } else {
+        push(@curlopt, "silent");
+        push(@curlopt, "insecure") if ($use_ssl && $is_get_ip);  ## this should really be a user option
```

Comment:
IIUC, IO::Socket::SSL does validation by default nowadays so this is a security regression.

Reply:
Happy to remove this if it is consistent with how you handle it in IO::Socket::SSL implementation.  For context we first implemented curl support several years ago when ddclient did not have its own SSL and when websites may not have properly signed certificates.

Reply:
Yes, remove it.

Reply:
removed